### ISSUE TITLE
Problem with same name sections

### ISSUE THREAD

**Issue by Krakean** [cemdervis/SharpConfig#28]
Hello.
If we try to save this file with SharpConfig:

[System]
LoggerEnabled=True
LoggerVerbosity=5
MasterServerIP=127.0.0.1
ListenPort=4433

[DEFAULT]
ConnectionType=initiator
UseDataDictionary=Y
ResetOnLogon=Y
ResetOnLogout=Y
ResetOnDisconnect=Y
HeartBtInt=30
ReconnectInterval=30

[SESSION]
BeginString=FIX.4.4

[SESSION]
BeginString=FIX.4.5

We will get mostly empty file as result:
[System]
LoggerEnabled = True
LoggerVerbosity = 5
MasterServerIP = 193.124.113.36

Can you please fix this issue?

**Comment by cemdervis** [comment]
Hello Krakean,
I will look into the problem shortly. As you pointed out, it's because of sections that have the same name. Does the problem still exist if you rename the second SESSION to something else?

**Comment by Krakean** [comment]
Nope, this issue exists only in case of same-named sections.

P.S. Would be nice if you will, besides fixing saving, also add functions like GetSections with ability to access to its settings. But fix saving comes first :)

**Comment by cemdervis** [comment]
What do you mean by GetSections()? You can already iterate through all sections and their settings.

**Comment by Krakean** [comment]
>You can already iterate through all sections and their settings

Even if they're same-named? What method I can use to do this?
P.S. Then all have to do is just fix saving :)

**Comment by cemdervis** [comment]
The Configuration class has static properties (IgnoreDuplicateSections and IgnoreDuplicateSettings).
If you set them to false, SharpConfig will parse everything that is in the configuration.
Then you just loop through your config instance with
```csharp
for (Section s in cfg)
    if (s.Name == "SESSION")
        // ...
```

But I will remove these properties, so SharpConfig will allow duplicate sections and settings by default, starting with the next release. These properties cause more confusion than necessary.
To then iterate through all sections with the same name, there will be a method such as:
```csharp
for (Section s in cfg.GetSectionsNamed("SESSION"))
    // ...
```

**Comment by Krakean** [comment]
Nice! Waiting for these changes :)
But please, if you can, release saving fix first, because its crucial for me right now :(

**Comment by cemdervis** [comment]
It's already fixed, just working on some API changes now, and preparing the release. Stay tuned!

**Comment by cemdervis** [comment]
Alright, both issues are fixed with the new version (1.5.5). Cheers!
